arlo-client/src/components/Audit/useSetupMenuItems/getJurisdictionFileStatus.ts

```diff
@@ -0,0 +1,56 @@
+import { api, checkAndToast } from '../../utilities'
+import { IErrorResponse } from '../../../types'
+
+type IJurisdictionsFileResponse =
+  | { file: null; processing: null }
+  | {
+      file: {
+        contents: null | string
+        name: string
+        uploadedAt: string
+      }
+      processing:
+        | {
+            status: 'READY_TO_PROCESS'
+            startedAt: null
+            completedAt: null
+            error: null
+          }
+        | {
+            status: 'PROCESSING'
+            startedAt: string
+            completedAt: null
+            error: null
+          }
+        | {
+            status: 'PROCESSED'
+            startedAt: string
+            completedAt: string
+            error: null
+          }
+        | {
+            status: 'ERRORED'
+            startedAt: string
+            completedAt: string
+            error: string
+          }
+    }
+
+const getJurisdictionFileStatus = async (
+  electionId: string
+): Promise<
+  'ERRORED' | 'PROCESSING' | 'PROCESSED' | 'ERRORED' | 'READY_TO_PROCESS'
+> => {
+  const jurisdictionsOrError:
+    | IJurisdictionsFileResponse
+    | IErrorResponse = await api(`/election/${electionId}/jurisdiction/file`)
+  if (checkAndToast(jurisdictionsOrError)) {
+    return 'ERRORED'
+  }
+  if (jurisdictionsOrError.processing) {
+    return jurisdictionsOrError.processing!.status
+  }
+  return 'READY_TO_PROCESS'
+}
+
+export default getJurisdictionFileStatus
```

Comment:
```suggestion
import { api, checkAndToast } from '../../utilities'
import { IErrorResponse } from '../../../types'

enum FileProcessingStatus {
  ReadyToProcess = 'READY_TO_PROCESS',
  Processing = 'PROCESSING',
  Processed = 'PROCESSED',
  Errored = 'ERRORED',
}

type IJurisdictionsFileResponse =
  | { file: null; processing: null }
  | {
      file: {
        contents: null | string
        name: string
        uploadedAt: string
      }
      processing:
        | {
            status: FileProcessingStatus.ReadyToProcess
            startedAt: null
            completedAt: null
            error: null
          }
        | {
            status: FileProcessingStatus.Processing
            startedAt: string
            completedAt: null
            error: null
          }
        | {
            status: FileProcessingStatus.Processed
            startedAt: string
            completedAt: string
            error: null
          }
        | {
            status: FileProcessingStatus.Errored
            startedAt: string
            completedAt: string
            error: string
          }
    }

const getJurisdictionFileStatus = async (
  electionId: string
): Promise<FileProcessingStatus> => {
  const jurisdictionsOrError:
    | IJurisdictionsFileResponse
    | IErrorResponse = await api(`/election/${electionId}/jurisdiction/file`)
  if (checkAndToast(jurisdictionsOrError)) {
    return FileProcessingStatus.Errored
  }
  if (jurisdictionsOrError.processing) {
    return jurisdictionsOrError.processing.status
  }
  return FileProcessingStatus.ReadyToProcess
}

export default getJurisdictionFileStatus
```

Reply:
@eventualbuddha I incorporated the enum bit, but for some reason the union type implementation was breaking things. It was acting as if only the last one was valid, and so it only let me use `'ERRORED'` as a value for `status`. I redid it a bit differently, and it works, let me know if you know of a better way, though.